### ISSUE TITLE
Performance regression?

### ISSUE THREAD

**Issue by mcfarljm** [dwavesystems/dwave-greedy#23]
I'm seeing a performance regression in steepest descent in Ocean 4.1 vs Ocean 3.5.  With Ocean 3.5, the sample code runs in about 2.4s, whereas with Ocean 4.1 it runs in about 3.9s.  Is this expected?  From what I can tell, they're both using the same version of `greedy`, so it seems surprising.

```python
from time import perf_counter

import dimod
import greedy

bqm = dimod.generators.anti_crossing_clique(150)
sampler = greedy.SteepestDescentSolver()

def solve(bqm, sampler):
    results = sampler.sample(bqm, num_reads=50000)
    return results.first

tic = perf_counter()
first = solve(bqm, sampler)
toc = perf_counter()

print(toc-tic)
```

(Use of `anti_crossing_clique` as the test BQM is not significant, I just wanted something deterministic.  I also saw similar differences with randomly generated BQM's.)

**Comment by randomir** [comment]
Absolutely. We've being toying with this idea for a while now, but with CQM out, we might want to prioritize it.

**Comment by mcfarljm** [comment]
Appears that this has been resolved in Ocean 5.0.
